### PR TITLE
Explicitly list the EC2 instance families that we will use in CI

### PR DESCRIPTION
**Pull request by zackgalbreath**

### CODE REVIEW

**Comment by AlmightyYakob** [comment]
It seems like that list of families must be almost all of them. Would it be easier to just exclude the ones that aren't allowed?

**Comment by zackgalbreath** [comment]
> It seems like that list of families must be almost all of them. Would it be easier to just exclude the ones that aren't allowed?

Good point, that does make for a shorter list.

File: k8s/karpenter/provisioners.yaml [hunk 1]
```diff
@@ -17,11 +17,40 @@ spec:
       memory: 20Ti
 
   requirements:
-    # Only spin up arm64 nodes
+    # Only spin up arm64 and amd64 nodes
     - key: "kubernetes.io/arch"
       operator: In
       values: ["arm64", "amd64"]
 
+    # Don't provision nodes from certain instance families.
```
Comment (AlmightyYakob) [inline k8s/karpenter/provisioners.yaml:25]: Sorry to be a stickler, but just so we have a record of why we're doing this
```suggestion
    # Don't provision nodes from certain instance families.
    # These are the instance families that don't support x86_v3
```

**Review by AlmightyYakob — DISMISSED** [review]
Left one suggestion but otherwise looks good

**Review by AlmightyYakob — APPROVED** [review]

**Comment by zackgalbreath** [comment]
For posterity's sake, here's some `archspec` data courtesy of #339 

[x86_64_instance_uarch.csv](https://github.com/spack/spack-infrastructure/files/9873642/x86_64_instance_uarch.csv)

**Pull request merged**